### PR TITLE
Hotfix: compilation issue, target Android API level 31

### PR DESCRIPTION
**Pull request by OlegMakarenko**
**Problem**
1.  Compilation issue descibed here: https://github.com/facebook/react-native/issues/35210
2. Warning from Android Play Store: 

> App not available to new users on Android versions higher than your target API level
> App must target Android 12 (API level 31) or higher

**Solution**
- Updated configuration in `build.gradle`.
- Bump SDK version to 31.
- Declare the android:exported in the activities that use intent filters.
- Remove unused dependency that caused upgrade issue (react-native-background-fetch).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ This application is built on `React Native` and therefore assumes you have all t
 ### Prerequisites
 
 - The **default network configuration** is fetched from the `env/default.json` file, and as part of the first-time setup, you should provide these. Check the `env/default.json.example` file to see what keys are required.
-- Node version should be **12.1.0** or above.
+- Node version should be from **12.1.0** to **17.9.1**.
```
Comment (Wayonb) [inline README.md:13]: Node 12 is really old.  I would increase the node version from 12.1.0 to 16.x if possible.
Even though node 16 will not be supported in a couple of weeks.
``from **16.x** to **17.9.1**.``

File: android/build.gradle [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
-        targetSdkVersion = 30
+        compileSdkVersion = 31
+        targetSdkVersion = 31
         supportLibVersion = "30.0.0"
         RNNKotlinVersion = "1.3.61" // Or any version above 1.3.x
         RNNKotlinStdlib = "kotlin-stdlib-jdk8"
+
+        if (System.properties['os.arch'] == "aarch64") {
+            // For M1 Users we need to use the NDK 24 which added support for aarch64
+            ndkVersion = "24.0.8215888"
```
Comment (Wayonb) [inline android/build.gradle:16]: Is there a reason not to use this ``ndkVersion``  as the minimum for all platforms?

**Review by Wayonb — CHANGES_REQUESTED** [review]

**Comment by OlegMakarenko** [comment]
@Wayonb thanks, fixed.

**Review by Wayonb — APPROVED** [review]
👍🏾

**Review by AnthonyLaw — APPROVED** [review]

**Pull request merged**